src/userAgent.ts

```diff
@@ -0,0 +1,22 @@
+import { AxiosHeaders } from "axios";
+import { ExtendedAxiosRequestConfig } from "./NetworkProviderConfig";
+
+export function setUserAgent(userAgentPrefix: string, config: ExtendedAxiosRequestConfig | undefined) {
+    if (!config) {
+        config = { headers: new AxiosHeaders({}) }
+    }
+    if (!config.headers) {
+        config.headers = new AxiosHeaders({})
+    };
+
+    const headers = AxiosHeaders.from(config.headers as AxiosHeaders).normalize(true);
+    if (!config.clientName) {
```

Comment:
Check can be above `const headers = ...` (so that we have the checks nicely grouped above).

Reply:
done